Sdtrig.tex

```diff
@@ -319,6 +319,37 @@ \subsection{Cache Operations}
     \end{steps}
 \end{commentary}
 
+\subsection{Invalid Virtual Addresses}
+
+For address matches without a mask, \RcsrTdataTwo must be able to hold all valid
+addresses in all supported translation modes. That means that after writing any
+of these valid addresses, the exact same value is read back, including any high
+bits.
+
+\begin{commentary}
+    If physical addresses are less than XLEN bits wide, they are zero-extended.
+    If virtual addresses are less than XLEN bits wide, they are sign-extended.
+    Be sure to have enough flops in \RcsrTdataTwo to hold all XLEN bits of the
+    full range of both physical and virtual addresses.
+\end{commentary}
+
+If \RcsrTdataTwo can hold any invalid addresses, then writes of an
+invalid address that can not be represented as-is should be converted to
+a different invalid address that can be represented.
+
+For invalid instruction fetch and load and store effective addresses,
+the compare value may be changed to a different invalid address.
+
+\begin{commentary}
+    A straightforward trigger implementation would compare the instruction fetch
+    and load and store effective addresses to the contents of \RcsrTdataTwo.
+    This is most intuitive to a user setting triggers. Here we accommodate
+    implementations that internally shorten invalid addresses.
```

Comment:
"Internally shorten" is not a properly (i.e. explicitly) defined term in this spec; one just hopes that the reader interpets these words exactly as intended.  Plus this is an implementation-level description in what is supposed to be an architecture-level spec.

What I think what is trying to be said is that invalid addresses, when written to tdata2, may be remapped to and stored as a supported invalid address value (if there are any supported invalid address values).  Which has already been said above normatively.

Net, it seems like this note can/should be removed.

Reply:
I'm removing this commentary.